src/index.js

```diff
@@ -0,0 +1,92 @@
+/** @import { Envelope, OfflineStore, Transport, OfflineTransportOptions } from '@sentry/core' */
+/** @import { Statement, Database } from 'better-sqlite3' */
+
+/** @typedef {Parameters<typeof makeNodeTransport>[0]} NodeTransportOptions */
+/**
+ * @typedef {Omit<OfflineTransportOptions, 'createStore'> & NodeTransportOptions & {
+ *   db: Database
+ *   table?: string
+ * }} OfflineSqliteTransportOptions
+ */
+
+/** @typedef {{time: number, envelope: string | Uint8Array, id: number}} Record */
+
+import { makeNodeTransport } from '@sentry/node'
+import {
+  makeOfflineTransport,
+  parseEnvelope,
+  serializeEnvelope,
+} from '@sentry/core'
+export const DEFAULT_TABLE = 'sentry_envelopes'
+/**
+ * @param {OfflineSqliteTransportOptions} opts
+ * @returns {Transport}
+ */
+export function makeOfflineSqliteTransport({
+  db,
+  table = DEFAULT_TABLE,
+  ...opts
+}) {
+  return makeOfflineTransport(makeNodeTransport)({
+    createStore: () => new SqliteOfflineStore(db, table),
+    ...opts,
+  })
+}
+export class SqliteOfflineStore {
```

Comment:
This will just add a little bit of type safety - it will catch errors in the type definitions of the methods here.
```suggestion
/** @implements {OfflineStore} */
export class SqliteOfflineStore {
```